unit-tests/src/test/scala/java/util/PropertiesTest.scala

```diff
@@ -0,0 +1,448 @@
+/**
+ * Ported from Harmony
+ */
+package java.util
+
+import java.io._
+import java.{util => ju}
+
+import org.junit.Test
+import org.junit.Ignore
+import org.junit.Assert._
+import org.junit.Assume._
+
+import scala.scalanative.junit.utils.AssertThrows._
+import scala.scalanative.junit.utils.Utils._
+
+class PropertiesTest {
+  // remove when Platform is implemented
+  val hasCompliantAsInstanceOfs = true
+
+  // ported from Scala.js - 2 tests ignored below
+  @Test def setProperty(): Unit = {
+    val prop = new Properties()
+    prop.setProperty("a", "A")
+    assertEquals("A", prop.get("a"))
+    prop.setProperty("a", "AA")
+    prop.setProperty("b", "B")
+    assertEquals("AA", prop.get("a"))
+    assertEquals("B", prop.get("b"))
+
+    val prop2 = new Properties(prop)
+    prop2.setProperty("a", "AAA")
+    assertEquals("AAA", prop2.get("a"))
+  }
+
+  @Test def getProperty(): Unit = {
+    val prop = new Properties()
+
+    assertNull(prop.getProperty("a"))
+    prop.setProperty("a", "A")
+    assertEquals("A", prop.getProperty("a"))
+    assertNull(prop.getProperty("aa"))
+
+    assertEquals("A", prop.getProperty("a", "B"))
+    assertEquals("B", prop.getProperty("b", "B"))
+
+    // Tests with default properties
+    prop.setProperty("b", "B")
+
+    val prop2 = new Properties(prop)
+    prop2.setProperty("b", "BB")
+    prop2.setProperty("c", "C")
+    assertEquals("A", prop2.getProperty("a"))
+    assertEquals("BB", prop2.getProperty("b"))
+    assertEquals("C", prop2.getProperty("c"))
+  }
+
+  @Test def propertyNames(): Unit = {
+    val prop = new Properties()
+    assertTrue(enumerationIsEmpty(prop.propertyNames()))
+    prop.setProperty("a", "A")
+    prop.setProperty("b", "B")
+    prop.setProperty("c", "C")
+    assertEquals(3, enumerationSize(prop.propertyNames()))
+    assertEnumSameElementsAsSet[Any]("a", "b", "c")(prop.propertyNames())
+
+    val prop2 = new Properties(prop)
+    prop.setProperty("c", "CC")
+    prop.setProperty("d", "D")
+    assertEquals(4, enumerationSize(prop2.propertyNames()))
+    assertEnumSameElementsAsSet[Any]("a", "b", "c", "d")(prop2.propertyNames())
+  }
+
+  @Ignore @Test def propertyNamesIsNotAffectedByOverriddenPropertyNamesInDefaults()
```

Comment:
Something seems to be wrong with `default` handling - these tests are from Scala.js.

Reply:
These two tests do not pass, I overlooked these when porting from Scala.js. They have to do with the default properties. If you see something obvious let me know. I need to fix these two and remove `@ignore`.

Reply:
Resolved by using code from Scala.js and porting `ScalaOps` over.

Reply:
Ah, but all that new code has nothing to do with this PR :s

Could you set aside that last commit, remove it from this PR, and resubmit it as a separate PR once this one is merged please? And instead, *delete* the `@Ignored` tests in this PR, since they have nothing to do with `load` and `store` either. They should be part of the separate PR that focuses on fixing `propertyNames` and `stringPropertyNames`.

Reply:
Sure thing - Issue created: https://github.com/scala-native/scala-native/issues/1871